src/dawn/Optimizer/StencilInstantiation.h

```diff
@@ -563,6 +566,13 @@ class StencilInstantiation : NonCopyable {
     return BoundaryConditionToExtentsMap_.find(stmt)->second;
   }
 
+  Array3i getFieldIDFromInitializedDimensionsMap(int FieldID) {
+    if(fieldIDToInitializedDimensionsMap_.count(FieldID) == 0) {
+      return Array3i{{1, 1, 1}};
```

Comment:
why if not found it returns {1,1,1}  ? this is not intuitive

Reply:
because then it is a temporarily allocated field that should be allowed to be accessed in any dimension

Reply:
ok, but I think this is very hidden and unclear here. Does the fact that fieldID is not registered automatically assumes it is temporary? this can be error prone and this logic should not be in a getter, but rather somewhere where we do explicitely initialize the temporaries:

```
for(auto tmp: allTemps) {
  setInitializedDimension(fieldID, Array3i{1,1,1});
}
```